src/main/java/seedu/commands/EditCommand.java

```diff
@@ -52,17 +54,17 @@ public CommandResult execute(ExpenditureList expenditures) {
             editExpenditure(editedExpenditure, isLendOrBorrowExpenditure);
             handleLumpSumExpenditure(editedExpenditure, isLendOrBorrowExpenditure);
             handleLendBorrowExpenditure(editedExpenditure);
+            ExceptionChecker.checkDate(fetchDate(isLendOrBorrowExpenditure), fetchDeadline());
```

Comment:
checkDate method should be placed in handleLendBorrowExpenditure as the code will run fetchDeadline() for regular expenditure categories and produce an exception